infrastructure/logging/log_summarizer.cc

```diff
@@ -0,0 +1,67 @@
+//%bin(log_summarizer)
+
+//%deps(yaml-cpp)
+#include <iostream>
+#include <iomanip>
+#include "infrastructure/logging/log_reader.hh"
+#include "infrastructure/config/config.hh"
+#include "infrastructure/time/timestamp.hh"
+#include "embedded/imu_driver/imu_message.hh"
+
+
+struct ChannelStats{
+  jet::Timestamp first_message_received;
+  jet::Timestamp last_message_received;
+  int num_messages_received;
+  double frequency;
+};
+
+std::optional<std::string> parse_commandline_arguments(int argc, char* argv[]) {
+  if (argc > 1) {
+    return argv[1];
+  }
+  return {};
+}
+
+int main(int argc, char* argv[]) {
+  const std::optional<std::string> log_path = parse_commandline_arguments(argc, argv);
+  Config log_config;
+  if (!log_path.has_value()) {
+    std::cout << "Please enter a log path" << std::endl;
+    return 0;
+  }
+
+  jet::LogReader log_reader(*log_path, {});
```

Comment:
Consider making a function that takes a path to a log as an argument, and returns a summary